### PR TITLE
Create unified structure for all support packages

### PR DESCRIPTION
**Pull request by Svastits**
- use only lower case letters in resources

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # kuka_robot_descriptions
 
-This package contains resources that can be used with real KUKA robots as well as with simulations.
+This package contains support packages that can be used with real KUKA robots as well as with simulations.
```
Comment (gavanderhoorn) [inline README.md]:
```suggestion
This repository contains support packages that can be used with real KUKA robots as well as with simulations.
```

File: kuka_agilus_support/launch/test_kr6_r700-sixx.launch.py [hunk 2]
```diff
@@ -25,16 +25,17 @@ def generate_launch_description():
             " ",
             PathJoinSubstitution(
                 [FindPackageShare("kuka_agilus_support"),
-                 "urdf", "kr6_r700_sixx.urdf.xacro"]
+                 "urdf", "kr6_r700-sixx.urdf.xacro"]
```
Comment (gavanderhoorn) [inline kuka_agilus_support/launch/test_kr6_r700-sixx.launch.py]: while IIRC, legal, using dashes in filenames is not according to conventions.

Underscores tend to be used everywhere.

File: kuka_agilus_support/urdf/kr6_r700-sixx_macro.xacro [hunk 3]
```diff
@@ -5,118 +5,116 @@
   <xacro:include filename="$(find kuka_agilus_support)/urdf/kr_agilus_ros2_control.xacro" />
   <xacro:include filename="$(find kuka_agilus_support)/urdf/kr_agilus_transmission.xacro" />
 
-  <xacro:macro name="kuka_kr6_r700_sixx_robot" params="prefix client_ip client_port use_fake_hardware">
+  <xacro:macro name="kuka_kr6_r700-sixx_robot" params="prefix use_fake_hardware">
```
Comment (gavanderhoorn) [inline kuka_agilus_support/urdf/kr6_r700-sixx_macro.xacro]: While this name is technically not a name of a resource (in the ROS sense), I would also recommend against using dashes here.

Dashes in resource names are not legal (see [wiki/Names](http://wiki.ros.org/Names#Valid_Names), I don't believe there is a ROS 2 version of that page yet). Not using them in macro names would just make it easier for a user to guess or assume the spelling.

File: kuka_lbr_iisy_support/urdf/lbr_iisy3_r760_macro.xacro [hunk 4]
```diff
@@ -1,21 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="LBRiisy3R760">
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="lbr_iisy3_r760">
```
Comment (gavanderhoorn) [inline kuka_lbr_iisy_support/urdf/lbr_iisy3_r760_macro.xacro:2]: :+1:

File: kuka_agilus_support/launch/test_kr6_r700-sixx.launch.py (hunk 2 above)
Comment (Svastits) [reply kuka_agilus_support/launch/test_kr6_r700-sixx.launch.py]: Thanks, I will correct this in all places

File: kuka_lbr_iiwa_moveit_config/srdf/lbr_iiwa_arm.macro.xacro [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:macro name="kuka_lbr_iiwa_arm" params="prefix">
+    <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
+    <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
+    <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
+    <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
+    <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
+    <group name="lbr_iiwa_arm">
+      <chain base_link="${prefix}base_link" tip_link="${prefix}flange" />
+    </group>
```
Comment (gavanderhoorn) [inline kuka_lbr_iiwa_moveit_config/srdf/lbr_iiwa_arm.macro.xacro]: I just happen to notice it in this `.srdf`, and it's really just something to consider, but at least in ROS-Industrial, we've tried to always make the default `chain` run from `base_link` to `tool0` and call it `manipulator`.

FK/IK would then probably match with what the TP shows, and by using the generic `manipulator` name, switching between manipulators (even from the same OEM) would not require updating references 'everywhere'.

File: kuka_robot_descriptions/package.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>kuka_robot_descriptions</name>
+  <version>0.0.1</version>
+  <description>Robot desciptions for KUKA robots</description>
+  <author>Áron Svastits</author>
+  <maintainer email="svastits1@gmail.com">Áron Svastits</maintainer>
+  <license>BSD</license>
```
Comment (gavanderhoorn) [inline kuka_robot_descriptions/package.xml:8]: Suggestion: switch to using [SPDX license identifiers](https://spdx.org/licenses).

They are machine readable, easier to verify (by automated tools) and leave no room for ambiguity (*which* variant of BSD is this fi?).

**Comment by gavanderhoorn** [comment]
Apologies for the many comments. I should've submitted a review instead.

File: kuka_lbr_iiwa_moveit_config/srdf/lbr_iiwa_arm.macro.xacro (hunk 5 above)
Comment (Svastits) [reply kuka_lbr_iiwa_moveit_config/srdf/lbr_iiwa_arm.macro.xacro]: good practice, I will fix this too :)

File: kuka_robot_descriptions/package.xml (hunk 6 above)
Comment (Svastits) [reply kuka_robot_descriptions/package.xml:8]: Initially this repo was ported from ros-industrial, where the support packages had BSD licenses, and we did not want to modify them. 
Thanks for your suggestion, we will consider using SPDX - there should be a licensing decision for the whole kroshu organization in the (close) future, after which we will update all licenses 
So my preferred way would be to create an issue, write down your suggestion and we will come back to licensing in a later PR

**Comment by Svastits** [comment]
> Apologies for the many comments. I should've submitted a review instead.

Thanks for the quick review, it was perfect this way too :)
If you do not have any other comments (aside from licensing, which decision I would postpone), I will merge this PR

File: kuka_robot_descriptions/package.xml (hunk 6 above)
Comment (gavanderhoorn) [reply kuka_robot_descriptions/package.xml:8]: > Initially this repo was ported from ros-industrial, where the support packages had BSD licenses, and we did not want to modify them.

in that case all `BSD` license references would be `BSD-3-Clause`.

> Thanks for your suggestion, we will consider using SPDX - there should be a licensing decision for the whole kroshu organization in the (close) future, after which we will update all licenses So my preferred way would be to create an issue, write down your suggestion and we will come back to licensing in a later PR

Sure. I just wanted to refer you to them.

re: licensing in general: `Apache-2.0` and `MIT` would be fully compatible with `BSD-3-Clause`.

**Comment by gavanderhoorn** [comment]
> If you do not have any other comments (aside from licensing, which decision I would postpone), I will merge this PR

I've not gone over everything, but for now I believe the majority has been addressed.

I'll post issues if I believe there is something else to suggest.

**Pull request merged**